File: system/src/main/java/com/cys/system/common/service/CartService.java
package com.cys.system.common.service;

import com.cys.system.common.common.pojo.Result;
import com.cys.system.common.exception.UnauthorizedException;


public interface CartService {
    Result addCart(Integer productId, Integer productNum, Integer userId);

    Result lookCart(Integer userId);

    Result deleteCartItem(Integer cartId,Integer[] ids,Integer userId) throws UnauthorizedException;

    Result clearCart(Integer userId);

    Result addNum(Integer cartId,Integer cartItemId,Integer num,Integer userId) throws UnauthorizedException;
}

File: search/src/main/resources/static/search/js/service/cartService.js
app.service("cartService", function ($http) {
    this.addCart = function (productId,num) {
        return $http.get(basePath+"/cart/add?productId="+productId+"&num="+num);
    };
    this.lookCart = function () {
        return $http.get(basePath+"/cart/look");
    };
    this.addNum = function(cartId,cartItemId,num){
        return $http.get(basePath+"/cart/addNum?cartId="+cartId+"&cartItemId="+cartItemId + "&num="+num);
    }
    this.deleteCartItem = function (cartId,ids) {
        return $http.delete(basePath+"/cart/delete?cartId="+cartId+"&ids="+ids);
    };
    this.clearCart = function () {
        return $http.get(basePath+"/cart/clear");
    };
    this.buildOrder = function(ids){
        return $http.post(basePath+"/order/build",ids);
    };
    this.pay = function(order){
        return $http.post(basePath+"/order/pay",order);
    };
    this.lookPayOrder = function(){
        return $http.get(basePath+"/order/lookPay");
    };
    this.lookNoPayOrder = function(status){
        return $http.get(basePath+"/order/lookNoPay?status="+status);
    };
    this.deleteOrder = function(orderId){
        return $http.get(basePath+"/order/delete?orderId="+orderId);
    }
});